apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/RetriableTransaction.cs

```diff
@@ -105,6 +105,8 @@ async Task<TResult> CommitAttempt()
 
         internal async Task<TResult> ExecuteWithRetryAsync<TResult>(Func<Task<TResult>> fn, CancellationToken cancellationToken)
         {
+            // Note: the way that we sometimes use the recommended retry delay, and base
```

Comment:
What about a `Func<Exception, TimeSpan?> RecommendedBackoff` in `RetrySettings`.

In `RetryAttempt` what is now `JitteredBackoff` can become `DefaultJitteredBackoff` and we can add `RecommendedBackoff` (that is set in `ShouldRetry` using the function). The Backoff methods would use `RecommendedBackoff ?? DefaultJitteredBackoff`.

When building the sequence , after the `yield return` we use `RetryAttempt.RecommendeBackoff ?? backoff` to calculate the new backoff. Since we are past the `yield return`, `RetryAttempt.ShouldRetry` has already been executed and `RetryAttempt.RecommendedBackoff` has been set (if any).

I get this might be a little bit more complicated than what we may want right now.
And also, happy to do this later in another PR.

Reply:
I think that's a bit complicated for what's effectively a one-off situation. I'm a big fan of only putting things into common libraries when we see the same pattern several times.